### PR TITLE
Fixes issue #10 (RuleEditWindow: use pop-up menus).

### PR DESCRIPTION
**Pull request by owenca**
Fixes memory leaks in ActionView and TestView.
Uses group layout management when removing an ActionView.
Removes manual resizing functions from TestView and ActionView.
Removes unused messages.

### CODE REVIEW

**Comment by humdingerb** [comment]
I'm not sure center-aligning the add/remove buttons is a good idea. The window already looks like there are too many big buttons all over the place. If left or right aligned, these buttons at least stick to the same line all the other buttons...

Maybe using right-aligned +/- buttons like the Repositories preferences would be better still.
Also, converting buttons to BPopUpMenus would tighten the GUI.

**Comment by owenca** [comment]
I will work on converting the buttons to menus. Would centered Add/Remove buttons look better then? They are centered in other windows.

**Comment by humdingerb** [comment]
Let's see how it looks once you've converted stuff to BPopMenus. May very well be that then centering the two remaining buttons is fine. Now, there are just too many buttons... :)

**Comment by owenca** [comment]
I have taken out the code that centers the Add/Remove buttons. Please see the new commit.

I have been trying to convert the test buttons to menus using BMenuField but can't make it work yet. It's much harder than I expected.

**Comment by humdingerb** [comment]
I think I still like the Add/Remove buttons better right aligned, as it distinguishes better what widgets are part of a "When" condition or "Do" action. That may become less relevant if you figure out the button-to-menu-conversion. :)

"BPopMenu" I mentioned above, is of course the wrong class...

Maybe take a look at the Find... panel in the Deskbar (attribute-mode) for code inspiration.

**Comment by owenca** [comment]
Please see the screenshots below:
left
center
right
The right-aligned design doesn't look good near the lower-right corner, where four buttons are crowded together. If you still think it's better than the other two, I will happily oblige.

**Comment by humdingerb** [comment]
I'll leave it up to you. :)
Maybe one day every row will get their own +/- widget (á là BSpinner) and these big buttons will be a thing of the past anyway. In any case, it's a matter of different tastes and no biggie at that.
Thanks for working on this!

**Comment by owenca** [comment]
No problem. I will leave the Add/Remove buttons where they were and make necessary changes after the misused buttons are converted to popup menus. Please see the latest commit which includes more cleanup of manual resizing code.

**Comment by owenca** [comment]
After many hours of frustration, I have finally figured out why the popup menu items' messages were not sent to ActionView once the action button in the Do box was converted to a BMenuField. Whew!

Please see the newly squashed commit above and the screenshot below:
action menu

**Comment by humdingerb** [comment]
I finally got around to checking it out. Seems to work. :)
Do you plan to BMenuFieldize the buttons in the "When" section as well?

**Comment by owenca** [comment]
I have been out of town vacationing with my family but will continue to work on converting the buttons in the When box as soon as I am back home. I tried to get it done before leaving, but they are much more involved than the button in the Do box. For example, when the "is" button is triggered, its popup menu is built on the fly based on the value of the "Type" button, which would also be converted to a BMenuField. I suspect that it's why BButton instead of BMenuField was used in the first place.

**Comment by humdingerb** [comment]
Maybe you can do it easier​ with one BMenuField with the "is", "is not" etc. in a submenu. See again the Find... panel.

**Comment by owenca** [comment]
Good idea! I will try it as last resort. One drawback is that the menu would be three layers deep. I also would have to deal with relabeling the menu bar of the menu field to "Type is", etc.

Meanwhile, I had an idea using BMenu::AddDynamicItem(add_state). It seems to work. Please see the newly squashed commit above.

**Comment by owenca** [comment]
The two buttons in the When box are now converted to BMenuField. Please see the commit above and the screenshot below:
popup menus

**Comment by humdingerb** [comment]
Very nice!
However, the "Do" action keeps reverting to "Move to folder..." after changing it to something else, pressing "OK" and re-opening it.

**Comment by owenca** [comment]
Fixed. Also cleaned up the code more. I think this is it for now.

File: sources/TestView.cpp [hunk 1]
```diff
@@ -173,73 +147,8 @@ TestView::MessageReceived(BMessage* msg)
...
-				point.y = 0.0;
-
-			menu->SetAsyncAutoDestruct(true);
-			menu->Go(point, true, true, true);
-			break;
-		}
-		case MSG_SHOW_MODE_MENU:
-		{
-			ShowModeMenu();
+			if (msg->FindString("mode",&mode) == B_OK)
```
Comment (humdingerb) [inline sources/TestView.cpp]: tiniest code style: space in front of &

Comment (owenca) [inline sources/TestView.cpp]: Fixed.

**Comment by humdingerb** [comment]
Thanks very much!

**Comment by smccrearyx** [comment]
Good choice in moving to BMenuField.  Looks much better that way.  Nice progress.

**Pull request merged**